src/Data/Bits/Extras.hs

```diff
@@ -38,22 +43,57 @@ import Data.Word
 import Foreign.Ptr
 import Foreign.Storable
 import GHC.Base
+#if __GLASGOW_HASKELL__ >= 900
+import Control.Exception
+import GHC.Integer.Logarithms
+#endif
 
 -- $setup
 -- >>> import Data.Bits (Bits(..))
 -- >>> import Data.Word (Word)
 
--- TODO: generalize to 64 bits, etc.
+-- | Calculate the integer base 2 logarithm. The argument must be strictly
+-- positive.
```

Comment:
It would be worth noting that the type of `log2` is more general in GHC 9.0+, as this is something that wouldn't be obvious when looking at the Haddocks alone.